File: apps/frontend/features/manager/user/services/user.service.ts
import api from "@/common/lib/axios";
import { ApiResponse } from "@/common/types/api.types";
import { ChangePasswordRequest, ResetFirstPasswordRequest, ResetPasswordRequest, UpdateUserRequest, UserRequest, UserResponse } from "../../user/types/user.types";

export const getAllUsers = async () => {
    const res = await api.get<ApiResponse<UserResponse[]>>("/users");
    return res.data;
};

export const getUserById = async (id: number) => {
    const res = await api.get<ApiResponse<UserResponse>>(`/users/${id}`);
    return res.data;
}
export const getCurrentUser = async () => {
    const res = await api.get<ApiResponse<UserResponse>>(`/users/me`);
    return res.data;
}

export const createUser = async (data: UserRequest) => {
    const res = await api.post<ApiResponse<void>>("/users", data);
    return res.data;
};

export const updateUser = async ({ id, data }: { id: number, data: UpdateUserRequest }) => {
    const res = await api.put<ApiResponse<void>>(`/users/${id}`, data);
    return res.data;
};

export const deleteUser = async ({ id }: { id: number }) => {
    const res = await api.delete<ApiResponse<void>>(`/users/${id}`);
    return res.data;
};

export const changePassword = async (data: ChangePasswordRequest) => {
    const res = await api.put<ApiResponse<void>>("/users/me/change-password", data);
    return res.data;
};

export const resetFirstPassword = async (data: ResetFirstPasswordRequest) => {
    const res = await api.put<ApiResponse<void>>("/users/me/reset-first-password", data);
    return res.data;
};

export const resetPassword = async ({ id, data, }: { id: number; data: ResetPasswordRequest; }) => {
    const res = await api.put<ApiResponse<void>>(`/users/${id}/reset-password`, data);
    return res.data;
};

File: apps/backend/src/main/java/com/supertech/superbatch/manager/user/service/UserService.java
package com.supertech.superbatch.manager.user.service;

import java.util.List;

import com.supertech.superbatch.manager.user.dto.ChangePasswordRequest;
import com.supertech.superbatch.manager.user.dto.ResetFirstPasswordRequest;
import com.supertech.superbatch.manager.user.dto.ResetPasswordRequest;
import com.supertech.superbatch.manager.user.dto.UpdateUserRequest;
import com.supertech.superbatch.manager.user.dto.UserRequest;
import com.supertech.superbatch.manager.user.dto.UserResponse;

public interface UserService {

    List<UserResponse> getAll();

    UserResponse getById(Long id);

    UserResponse getCurrentUser(Long currentUserId);

    void create(UserRequest request, Long userId);

    void update(Long id, UpdateUserRequest request);

    void delete(Long id, Long currentUserId);

    void resetFirstPassword(ResetFirstPasswordRequest request, Long currentUserId);

    void changePassword(ChangePasswordRequest request, Long currentUserId);

    void resetPassword(ResetPasswordRequest request, Long currentUserId);

}